robot_control_components/include/robot_control_components/simple_component.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 ROS2-Control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef ROBOT_CONTROL_COMPONENTS__COMPONENTS_COMPONENT_HPP_
+#define ROBOT_CONTROL_COMPONENTS__COMPONENTS_COMPONENT_HPP_
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "control_msgs/msg/interface_value.hpp"
+
+#include "robot_control_components/components/component.hpp"
+
+#include "robot_control_components/ros2_control_types.h"
+#include "robot_control_components/visibility_control.h"
+
+
+namespace robot_control_components
+{
+
+template < typename ComponentHardwareType >
+class SimpleComponent : public Component< ComponentHardwareType >
```

Comment:
i'd propose to leave out this class as for now and really only focus on sensors and actuators before going into templates and generic implementations.